src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -232,6 +233,55 @@ public TagHelperContent EndTagHelperWritingScope()
         }
 
         /// <summary>
+        /// Starts a new scope for writing <see cref="ITagHelper"/> attribute values.
+        /// </summary>
+        /// <remarks>
+        /// All writes to the <see cref="Output"/> or <see cref="ViewContext.Writer"/> after calling this method will
+        /// be buffered until <see cref="EndWriteTagHelperAttribute"/> is called.
+        /// The content will be buffered using a shared <see cref="StringWriter"/> within this <see cref="RazorPage"/> 
+        /// This method assumes that there will be no nesting of <see cref="BeginWriteTagHelperAttribute"/> 
```

Comment:
Rather than assuming anything, suggest asserting that `_pageWriter == null` here and resetting `_pageWriter` in `EndWriteTagHelperAttribute()`.